streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -962,6 +965,7 @@ private boolean close(final long timeoutMs) {
             return true;
         } else {
             log.info("Streams client cannot stop completely within the timeout");
+            Arrays.stream(threads).forEach(StreamThread::interrupt);
```

Comment:
Calling `Thread.interrupt` on a dead thread causes no issues.

Reply:
We also need to make sure that inside StreamThread all the blocking calls that can possibly throw interrupted exceptions now are gracefully handled since otherwise it may cause undefined behavior.

Reply:
Why the threads here are considered dead? They may be still in the middle of a runOnce right?

Reply:
Yes, you are correct.  But I'm not considering any threads dead at this point.  I wanted to proactively call attention to the fact it's safe to call `Thread#interrupt()`even on threads that **_may be_** in a dead state.  I admit the comment is somewhat unclear.